solver/src/solver/http_solver/model.rs

```diff
@@ -26,14 +26,16 @@ pub struct OrderModel {
 }
 
 #[derive(Debug, Serialize)]
-pub struct UniswapModel {
-    pub token1: String,
-    pub token2: String,
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    pub balance1: u128,
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    pub balance2: u128,
+pub struct PoolTokenData {
     #[serde(with = "serde_with::rust::display_fromstr")]
+    pub balance: u128,
+    pub weight: String,
+}
+
+#[derive(Debug, Serialize)]
+pub struct PoolModel {
+    pub pool_type: String,
```

Comment:
I think instead of using a `String` here we should have an `enum` that serializes to and from some string representation. This should be possible with `serde`.